common.py

```diff
@@ -1005,12 +1005,43 @@ def apply_rt_signalregion(cols):
     cols.cutflow['rt_signalregion'] = len(cols)
     return cols
 
+def check_if_model_exists(model_file, xrootd_url) :
+    # Check if the file exists locally
+    if not os.path.exists(model_file):
+        print(f"File {model_file} not found. Downloading from {xrootd_url}...")
+        try:
+            os.makedirs(os.path.dirname(model_file), exist_ok=True)  # Ensure directory exists
+            subprocess.run(["xrdcp", xrootd_url, model_file], check=True)
+            print(f"Downloaded {model_file} successfully.")
+        except subprocess.CalledProcessError as e:
+            print(f"Error downloading {model_file}: {e}")
```

Comment:
this should be a `RuntimeError` (execution should not continue if required files cannot be downloaded)

Reply:
Changed

Reply:
This change is not what I meant. Rather than just printing a warning message, execution of the program should stop entirely if a file cannot be downloaded. This line should be `raise RuntimeError(...)`.